src/igl/opengl/RenderPipelineState.cpp

```diff
@@ -106,7 +106,7 @@ Result RenderPipelineState::create() {
   if (!IGL_DEBUG_VERIFY(desc_.shaderStages->getType() == ShaderStagesType::Render)) {
     return Result(Result::Code::ArgumentInvalid, "Shader stages not for render");
   }
-  const auto& shaderStages = std::static_pointer_cast<ShaderStages>(desc_.shaderStages);
+  const auto& shaderStages = static_cast<ShaderStages*>(desc_.shaderStages.get());
```

Comment:
Remove a reference to a pointer.

Reply:
like this : 
const auto shaderStages = static_cast<ShaderStages*>(desc_.shaderStages.get());
?

Reply:
auto shaderStages = static_cast<ShaderStages*>(desc_.shaderStages.get());

Reply:
done

Reply:
If it is confusing, you can spell all the type names in full instead of using `auto`. For example, as in `const ShaderStages*` etc. 

Don't forget `const`.